kratos/utilities/constraint.h

```diff
@@ -0,0 +1,297 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Aditya Ghantasala
+//
+//
+
+#if !defined(CONSTRAINT_H)
+#define CONSTRAINT_H
+// System includes
+
+// project includes
+#include "includes/define.h"
+#include "includes/dof.h"
+#include "includes/node.h"
+#include "includes/kratos_flags.h"
+#include "geometries/geometry.h"
+#include "containers/constraint_equation.h"
+#include "containers/flags.h"
+#include "containers/variable.h"
+#include "containers/variable_component.h"
+#include "containers/vector_component_adaptor.h"
+#include "containers/variable_data.h"
+
+namespace Kratos
+{
+/** \brief Constraint
+	* A class that implements the interface for different constraints to be applied on a system.
+    */
+class MasterSlaveConstraint :  public IndexedObject, public Flags
+{
+
+  public:
+    /// Pointer definition of DataValueContainer
+    KRATOS_CLASS_POINTER_DEFINITION(MasterSlaveConstraint);
+    typedef std::size_t IndexType;
+    typedef Dof<double> DofType;
+    typedef std::vector< DofType::Pointer > DofsVectorType;
+    typedef Node<3> NodeType;
+    typedef PointerVectorSet<NodeType, IndexedObject> NodesContainerType;
+    typedef std::vector<std::size_t> EquationIdVectorType;
+
+    typedef double ConstantType;
+    typedef Matrix MatrixType;
+    typedef Vector VectorType;
+    typedef Kratos::Variable<double> VariableType;
+    typedef Kratos::VariableComponent<Kratos::VectorComponentAdaptor<Kratos::array_1d<double, 3>>> VariableComponentType;
+    ///@name Life Cycle
+    ///@{
+
+   /*
+    * Empty Constructor
+    */
+    MasterSlaveConstraint():IndexedObject(0), Flags()
+    {
+    }
+
+    /*
+    * Constructor by passing a vector of Master and slave dofs and corresponding Matrix and constant vector
+    */
+    MasterSlaveConstraint(IndexType Id, DofsVectorType& rMasterDofsVector,
+                                        DofsVectorType& rSlaveDofsVector,
+                                        MatrixType RelationMatrix,
+                                        VectorType ConstantVector):IndexedObject(Id), Flags()
+    {
```

Comment:
Why define the constructor with unused parameters?

Reply:
@msandre for now these are only prototypes of constructors as suggested by @RiccardoRossi . I am currently implementing those.

Reply:
ok